src/confint.jl

```diff
@@ -10,8 +10,8 @@ BasicConfInt(0.95)
 ```
 
 """
-struct BasicConfInt <: ConfIntMethod
```

Comment:
Does switching to a parametrised method improve the performance or is this change rather aesthetically?

Reply:
I haven't benchmarked every change in detail but in general type declarations should [improve the performance](https://docs.julialang.org/en/v1/manual/performance-tips/index.html#Type-declarations-1).

Reply:
NOTE: Parameterizing the type only improves performance if you're dealing with wrapper types (e.g., `AbstractFloat`). If `level` was declared as `Float64` then this wouldn't make a difference.

Reply:
Are you referring to my comment? Defining `level` as a field with a concrete type like `Float64` would improve the performance as well but, of course, be much more restrictive. Currently (on the master and develop branch) the implementation of `BasicConfInt` is identical to the example `MyStillAmbiguousType` in the [discussion about fields with abstract type](https://docs.julialang.org/en/v1/manual/performance-tips/index.html#Avoid-fields-with-abstract-type-1).

Reply:
But as the example in the documentation shows, just constructing `BasicConfInt` with a concrete type such as `BasicConfInt(0.95)` is problematic.

Reply:
> just constructing BasicConfInt with a concrete type such as BasicConfInt(0.95) is problematic.

I cleared missed something, is there some documentation that suggests we might want to use Float32?

Reply:
> I clearly missed something, is there some documentation that suggests we might want to use Float32?

Sorry, I don't understand your question. What do you mean? Would you like to define
```julia
struct BasicConfInt
     level::Float64
end
```
instead and are wondering why that would be restrictive? My point was that this would be much more restrictive than the current declaration `level::AbstractFloat`. As you say, someone might want to use `Float32` (e.g., it's the standard floating point number type on Windows 32bit, if I'm not mistaken), dual numbers, or some arbitrary user-defined number types. I'm not sure whether people would make use of that to a large extent, but when making the changes in this PR, defining
```julia
struct BasicConfInt{T<:AbstractFloat}
     level::T
end
```
was the only (?) way to get a composite type without abstract fields that still gives the user the same amount of freedom and flexibility in providing different types of `level` as in the current implementation on the master branch.